tools/stats_utils/s3_stat_parser.py

```diff
@@ -148,13 +149,42 @@ def _parse_s3_summaries(summaries: Any, jobs: List[str]) -> Dict[str, List[Any]]
 # Collect and decompress S3 test stats summaries into JSON.
 # data stored on S3 buckets are pathed by {sha}/{job} so we also allow
 # optional jobs filter
-def get_test_stats_summaries(*, sha: str, jobs: Optional[List[str]] = None) -> Dict[str, List[Any]]:
+def get_test_stats_summaries(*, sha: str, jobs: Optional[List[str]] = None) -> Dict[str, List[Report]]:
     bucket = get_S3_bucket_readonly(OSSCI_METRICS_BUCKET)
     summaries = bucket.objects.filter(Prefix=f"test_time/{sha}")
     return _parse_s3_summaries(summaries, jobs=list(jobs or []))
 
 
-def get_test_stats_summaries_for_job(*, sha: str, job_prefix: str) -> Dict[str, List[Any]]:
+def get_test_stats_summaries_for_job(*, sha: str, job_prefix: str) -> Dict[str, List[Report]]:
     bucket = get_S3_bucket_readonly(OSSCI_METRICS_BUCKET)
     summaries = bucket.objects.filter(Prefix=f"test_time/{sha}/{job_prefix}")
     return _parse_s3_summaries(summaries, jobs=list())
+
+
+# This function returns a list of S3 test time reports. This function can run into errors if HAVE_BOTO3 = False
+# or the S3 bucket is somehow unavailable. Even though this function goes through ten commits' reports to find a
+# non-empty report, it is still conceivable (though highly unlikely) for this function to return no reports.
+def get_previous_reports_for_branch(branch: str, ci_job_prefix: str = "") -> List[Report]:
+    commit_date_ts = subprocess.check_output(
+        ['git', 'show', '-s', '--format=%ct', 'HEAD'],
+        encoding="ascii").strip()
+    commit_date = datetime.fromtimestamp(int(commit_date_ts))
+    # We go a day before this current commit to avoiding pulling incomplete reports
+    day_before_commit = str(commit_date - timedelta(days=1)).split(' ')[0]
+    # something like git rev-list --before="2021-03-04" --max-count=10 --remotes="*origin/nightly"
+    commits = subprocess.check_output(
+        ["git", "rev-list", f"--before={day_before_commit}", "--max-count=10", f"--remotes=*{branch}"],
+        encoding="ascii").splitlines()
+
+    reports: List[Report] = []
+    commit_index = 0
+    while len(reports) == 0 and commit_index < len(commits):
+        commit = commits[commit_index]
+        print(f'Grabbing reports from commit: {commit}')
```

Comment:
nit: generally utility functions don't have prints. we can either remove it or convert it into debug message https://docs.python.org/3/library/logging.html